src/features/unified-share-modal/SharedLinkSection.js

```diff
@@ -51,11 +53,43 @@ type Props = {
     triggerCopyOnLoad?: boolean,
 };
 
-class SharedLinkSection extends React.Component<Props> {
+type State = {
+    autoCreatingSharedLink: boolean,
+};
+
+class SharedLinkSection extends React.Component<Props, State> {
     static defaultProps = {
         trackingProps: {},
+        autoCreateSharedLink: false,
     };
 
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            autoCreatingSharedLink: false,
+        };
+    }
+
+    componentDidUpdate(prevProps: Props) {
```

Comment:
Is there no possibility that this should handle the `autoCreateSharedLink` prop on mount as well as update? It's rare that one applies but not the other

Reply:
The lifecycle of the USM is such that the outer component will change the sharedlink and other props after a `getInitialData` has run. This component isn't responsible for creating the shared link, and all of that is delegated to the parent. in short:

* USM mounts, which will render this shared link component
* upon mount, USM will get initial data
* when it gets this data it will know whether it has a shared link or not. based on this, props get passed to the shared link component
* at that time, the component can tell if it should attempt to create one based on the update

@tjgupta I put it here, so that the shared link isn't added prior to the initial UI showing. If i put some of this in USM itself, it would wait until the shared link was generated to show the UI... this would prevent users from seeing the toggle slide, and/or the copied button from activating (eventually)

Reply:
Seems kind of fragile to rely on that order remaining the same. For example, the parent could just not render this component until it has data, then mount it with the props having the condition that needs this code to run.

Reply:
i am not going to use hooks for this. if you have a concrete reason why this would fail given how this component currently works, i'm all ears

Reply:
I'm not saying rewrite it. But there are no tests or documentation for this component that would indicate why this stops working if called in a slightly different way. It seems like the principle of least surprise would be that we handle this now, the same as if we do for components that could in theory only have componentDidMount and not handle prop updates.